File: Day024_2025/Q024_GFG.py
class Solution:
    # Function to check if the linked list has a loop.
    def detectLoop(self, head):
        if not head:
            return False

        slow = head
        fast = head

        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next

            if slow == fast:
                return True  # Loop detected

        return False  # No loop detected


# Definition for the Node class
class Node:
    def __init__(self, val):
        self.data = val
        self.next = None

File: Day024_2025/Q024_GFG.java
class Q024_GFG {
    // Function to check if the linked list has a loop.
    public static boolean detectLoop(Node head) {
        if (head == null) return false;

        Node slow = head;
        Node fast = head;

        while (fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;

            if (slow == fast) {
                return true; // Loop detected
            }
        }

        return false; // No loop detected
    }
}

// Definition for the Node class
class Node {
    int data;
    Node next;

    Node(int val) {
        data = val;
        next = null;
    }
}
